vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/api/SubmitWorkflowRequest.java

```diff
@@ -30,12 +30,11 @@ public boolean equals(Object o) {
       return false;
     }
     SubmitWorkflowRequest that = (SubmitWorkflowRequest) o;
-    return attempt == that.attempt && equalsIgnoreAttempt(that);
+    return attempt == that.attempt && Objects.equals(consumableResources, that.consumableResources) && equalsIgnoreAttemptAndConsumableResources(that);
   }
 
-  public boolean equalsIgnoreAttempt(SubmitWorkflowRequest that) {
+  public boolean equalsIgnoreAttemptAndConsumableResources(SubmitWorkflowRequest that) {
```

Comment:
Ignoring consumable resources in the hash and equals comparisons will trigger Shesmu to ignore actions generated by a change olive. 

The Shesmu scheduler does deduplication by placing actions as keys of a map, favouring a pre-existing action. This means that if an action is logically different, it needs to be a separate action. The attempt number is excluded from this because the olive does not set it. Although it occurs in the Vidarr submission request object, it is set to a default and then reset based on the Vidarr submission response. Consumable resources do not follow this pattern and so should not be changed in this manner. The reason this code lives in Vidarr rather than Shesmu is that, ultimately, if new fields are added, it's more likely that someone will remember to update the logic in the Vidarr codebase rather than the one in the Shesmu codebase.

Also, don't get the idea for it to prefer the new action. It _has_ to favour a pre-existing actions, or the actions would constantly be losing their state and most actions would end up in a permanent `UNKNOWN` limbo, as if Shesmu were locked in a reboot cycle.

While it may be irritating to have to purge actions after changing an olive, that is the design Shesmu requires. If you want to do otherwise, you'll need to redesign the Shesmu scheduler and action implementation.

Reply:
Thanks for the thorough comment and explanation!
We saw this in testing this weekend / this morning as well, so I've removed those changes.